tests/serverpod_test_server/test_e2e/websocket_auth_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Given a websocket that requires authentication', () {
+    TestAuthKeyManager authKeyManager = TestAuthKeyManager();
+    late Client client;
+
+    setUp(() async {
+      client = Client(
+        serverUrl,
+        authenticationKeyManager: authKeyManager,
+      );
+    });
+
+    tearDown(() async {
+      client.close();
+    });
+
+    test('then access restricted endpoint without authentication', () async {
```

Comment:
Suggestion(NIT): This test is missing it's `when` statement.

Same for the next test.